gulpfile.js

```diff
@@ -0,0 +1,870 @@
+}
+
+var sourceFiles = ['Source/**/*.js',
+                   '!Source/*.js',
+                   '!Source/Workers/**',
+                   '!Source/ThirdParty/Workers/**',
+                   'Source/Workers/createTaskProcessorWorker.js'];
+
+var buildFiles = ['Specs/**/*.js',
+                  '!Specs/SpecList.js',
+                  'Source/Shaders/**/*.glsl'];
+
+var jsHintFiles = ['Source/**/*.js',
+                   '!Source/Shaders/**',
+                   '!Source/ThirdParty/**',
+                   '!Source/Workers/cesiumWorkerBootstrapper.js',
+                   'Apps/**/*.js',
+                   'Apps/Sandcastle/gallery/*.html',
+                   '!Apps/Sandcastle/ThirdParty/**',
+                   'Specs/**/*.js',
+                   'Tools/buildTasks/**/*.js',
+                   'gulpfile.js'];
+
+var filesToClean = ['Source/Cesium.js',
+                    'Build',
+                    'Instrumented',
+                    'Source/Shaders/**.js',
+                    'Specs/SpecList.js',
+                    'Apps/Sandcastle/.jshintrc',
+                    'Apps/Sandcastle/jsHintOptions.js',
+                    'Apps/Sandcastle/gallery/gallery-index.js',
+                    'Cesium-*.zip'];
+
+var filesToSortRequires = ['Source/**/*.js',
+                           '!Source/Shaders/**',
+                           '!Source/ThirdParty/**',
+                           '!Source/Workers/cesiumWorkerBootstrapper.js',
+                           '!Source/copyrightHeader.js',
+                           '!Source/Workers/transferTypedArrayTest.js',
+                           'Apps/**/*.js',
+                           '!Apps/Sandcastle/ThirdParty/**',
+                           '!Apps/Sandcastle/jsHintOptions.js',
+                           'Specs/**/*.js',
+                           '!Specs/spec-main.js',
+                           '!Specs/SpecRunner.js',
+                           '!Specs/SpecList.js',
+                           '!Apps/Sandcastle/Sandcastle-client.js',
+                           '!Apps/Sandcastle/Sandcastle-header.js',
+                           '!Apps/Sandcastle/Sandcastle-warn.js',
+                           '!Apps/Sandcastle/gallery/gallery-index.js'];
+
+gulp.task('default', ['combine']);
+
+gulp.task('build', function(done) {
+    mkdirp.sync('Build');
+    glslToJavaScript(false, 'Build/minifyShaders.state');
+    createCesiumJs();
+    createSpecList();
+    createGalleryList();
+    createSandcastleJsHintOptions();
+    done();
+});
+
+gulp.task('build-watch', function() {
+    gulp.watch(buildFiles, ['build']);
+});
+
+gulp.task('buildApps', ['combine', 'minifyRelease'], function() {
+    return buildCesiumViewer();
+});
+
+gulp.task('clean', function(done) {
+    async.forEach(filesToClean, rimraf, done);
+});
+
+gulp.task('cloc', ['build'], function() {
+    var glsl = globby.sync(['Source/Shaders/*.glsl', 'Source/Shaders/**/*.glsl', 'Source/main.js']);
```

Comment:
`'Source/main.js'` doesn't really need to be fed through the glob thing.  It can just be included in the command line like the source directories are.